src/BenchmarkDotNet.TestAdapter/BenchmarkCaseExtensions.cs

```diff
@@ -0,0 +1,110 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Characteristics;
+using BenchmarkDotNet.Exporters;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Running;
+using Microsoft.TestPlatform.AdapterUtilities;
+using Microsoft.TestPlatform.AdapterUtilities.ManagedNameUtilities;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using System;
+using System.Linq;
+
+namespace BenchmarkDotNet.TestAdapter
+{
+    /// <summary>
+    /// A set of extensions for BenchmarkCase to support converting to VSTest TestCase objects.
+    /// </summary>
+    internal static class BenchmarkCaseExtensions
+    {
+        /// <summary>
+        /// Converts a BDN BenchmarkCase to a VSTest TestCase.
+        /// </summary>
+        /// <param name="benchmarkCase">The BenchmarkCase to convert.</param>
+        /// <param name="source">The dll or exe of the benchmark project.</param>
+        /// <param name="includeJobInName">Whether or not the display name should include the job name.</param>
+        /// <returns>The VSTest TestCase.</returns>
+        internal static TestCase ToVSTestCase(this BenchmarkCase benchmarkCase, string source, bool includeJobInName=false)
+        {
+            var benchmarkMethod = benchmarkCase.Descriptor.WorkloadMethod;
+            var fullClassName = benchmarkCase.Descriptor.Type.FullName;
```

Comment:
I am not sure whether this particular line to responsible for that, but for the generic types we display the number of generic type arguments in the benchmark name:

![image](https://github.com/dotnet/BenchmarkDotNet/assets/6011991/9c192bc8-4ae9-408f-98af-33aaa3d971f8)

You can use following method to get a nice output for all kinds of types (it caused us a lot of trouble in the past): https://github.com/dotnet/BenchmarkDotNet/blob/d49417314f6f361779e007042182f0a7ce13b45e/src/BenchmarkDotNet/Extensions/ReflectionExtensions.cs#L39